### DISCUSSION TITLE
pseudocode  code $$ vs $

### DISCUSSION THREAD

**Discussion by HaydarJawad** [alshedivat/al-folio#2460] [Q&A]
[pseudo.txt](https://github.com/alshedivat/al-folio/files/15474910/pseudo.txt)
Hi Guys, not sure if you came across this issue.  my pseudocode (belwo) works fine on $ on   pseudocode.js editor . when i add $$ (as per advice) I lose 50% of it mainly   the functions outside the curly brackets. any help appractetetd  i have been spending hours  with no success. Thank you 

% This algorithm checks the axioms of probability for an AI image classification model
\begin{algorithm}
\caption{Check Probabilities of Image Classification Model}
\begin{algorithmic}
\PROCEDURE{CheckProbabilities}{$sample\_space, event\_A, event\_B$}
    \STATE $total\_images \gets$ \CALL{Length}{$sample\_space$}
    \STATE $prob\_A \gets$ \CALL{Length}{$event\_A$} / $total\_images$
    \STATE $prob\_B \gets$ \CALL{Length}{$event\_B$} / $total\_images$

    \PROCEDURE{NonNegativity}{$prob$}
        \IF{$prob \geq 0$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \PROCEDURE{Normalisation}{$prob\_A, prob\_B$}
        \IF{$prob\_A + prob\_B = 1$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \PROCEDURE{Additivity}{$prob\_A, prob\_B$}
        \IF{$prob\_A + prob\_B \leq 1$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \STATE $is\_non\_negative \gets$ \CALL{NonNegativity}{$prob\_A$} \AND \CALL{NonNegativity}{$prob\_B$}
    \STATE $is\_normalised \gets$ \CALL{Normalisation}{$prob\_A, prob\_B$}
    \STATE $is\_additive \gets$ \CALL{Additivity}{$prob\_A, prob\_B$}

    \IF{$is\_non\_negative$}
        \STATE \textbf{print}("Non-negativity check passed")
    \ELSE
        \STATE \textbf{print}("Non-negativity check failed")
    \ENDIF

    \IF{$is\_normalised$}
        \STATE \textbf{print}("Normalisation check passed")
    \ELSE
        \STATE \textbf{print}("Normalisation check failed")
    \ENDIF

    \IF{$is\_additive$}
        \STATE \textbf{print}("Additivity check passed")
    \ELSE
        \STATE \textbf{print}("Additivity check failed")
    \ENDIF

    \STATE \textbf{print}("Probability of image being a cat: ", $prob\_A$)
    \STATE \textbf{print}("Probability of image being a dog: ", $prob\_B$)

\ENDPROCEDURE
\end{algorithmic}
\end{algorithm}

**Comment by george-gca** [comment]
I also tried to make it work, but to no avail. This is really weird.

**Comment by HaydarJawad** [comment]
Thank you  for trying, it's indeed  strange !

**Reply by george-gca** [reply]
I tested changing the MathJax version used to [tex-chtml-full](https://docs.mathjax.org/en/latest/web/components/combined.html#tex-chtml-full), as [stated in their docs](https://github.com/SaswatPadhi/pseudocode.js?tab=readme-ov-file#step-1c--for-mathjax-3x-users), which was the one difference I could spot, but to no avail.

```yml
  mathjax:
    integrity:
      js: "sha256-kbAFUDxdHwlYv01zraGjvjNZayxKtdoiJ38bDTFJtaQ="
    local:
      fonts: "output/chtml/fonts/woff-v2/"
    url:
      fonts: "https://cdn.jsdelivr.net/npm/mathjax@{{version}}/es5/output/chtml/fonts/woff-v2/"
      js: "https://cdn.jsdelivr.net/npm/mathjax@{{version}}/es5/tex-chtml-full.js"
    version: "3.2.2"
```

I believe it is worth opening an issue in [pseudocode repo](https://github.com/SaswatPadhi/pseudocode.js) asking for guidance in this.

**Reply by george-gca** [reply]
Also pasting it here so it is easier to copy:

```latex
% This algorithm checks the axioms of probability for an AI image classification model
\begin{algorithm}
\caption{Check Probabilities of Image Classification Model}
\begin{algorithmic}
\PROCEDURE{CheckProbabilities}{$sample\_space, event\_A, event\_B$}
    \STATE $total\_images \gets$ \CALL{Length}{$sample\_space$}
    \STATE $prob\_A \gets$ \CALL{Length}{$event\_A$} / $total\_images$
    \STATE $prob\_B \gets$ \CALL{Length}{$event\_B$} / $total\_images$

    \PROCEDURE{NonNegativity}{$prob$}
        \IF{$prob \geq 0$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \PROCEDURE{Normalisation}{$prob\_A, prob\_B$}
        \IF{$prob\_A + prob\_B = 1$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \PROCEDURE{Additivity}{$prob\_A, prob\_B$}
        \IF{$prob\_A + prob\_B \leq 1$}
            \STATE \textbf{return} True
        \ELSE
            \STATE \textbf{return} False
        \ENDIF
    \ENDPROCEDURE

    \STATE $is\_non\_negative \gets$ \CALL{NonNegativity}{$prob\_A$} \AND \CALL{NonNegativity}{$prob\_B$}
    \STATE $is\_normalised \gets$ \CALL{Normalisation}{$prob\_A, prob\_B$}
    \STATE $is\_additive \gets$ \CALL{Additivity}{$prob\_A, prob\_B$}

    \IF{$is\_non\_negative$}
        \STATE \textbf{print}("Non-negativity check passed")
    \ELSE
        \STATE \textbf{print}("Non-negativity check failed")
    \ENDIF

    \IF{$is\_normalised$}
        \STATE \textbf{print}("Normalisation check passed")
    \ELSE
        \STATE \textbf{print}("Normalisation check failed")
    \ENDIF

    \IF{$is\_additive$}
        \STATE \textbf{print}("Additivity check passed")
    \ELSE
        \STATE \textbf{print}("Additivity check failed")
    \ENDIF

    \STATE \textbf{print}("Probability of image being a cat: ", $prob\_A$)
    \STATE \textbf{print}("Probability of image being a dog: ", $prob\_B$)

\ENDPROCEDURE
\end{algorithmic}
\end{algorithm}
```

**Comment by HaydarJawad** [comment]
it works perfectly on

https://saswat.padhi.me/pseudocode.js/. if you copy and paste it.
only when you add $$ instead of $ it breaks down.

Regards
Haydar

**Reply by george-gca** [reply]
Yes, I know. But since they created the package, they might have an idea of what to do about it. Also we are using MathJax 3, which uses $$ instead of $, as [can be seen in its docs](https://docs.mathjax.org/en/latest/basic/mathematics.html#tex-and-latex-input), and what I did was just use the [setup recommended in pseudocode documentation](https://github.com/SaswatPadhi/pseudocode.js?tab=readme-ov-file#step-1c--for-mathjax-3x-users) to integrate it.

**Comment by HaydarJawad** [comment]
Got it, will release an issue there
Thank you
